sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CoreUtils.java

```diff
@@ -427,23 +429,51 @@ public static SerializationFormat serializationFormatFromContentType(HttpHeaders
     }
 
     /**
-     * Appends a query parameter to the given host URL.
+     * Appends a query parameter to the given URL.
      *
-     * @param url the base URL to which the query parameter will be appended.
-     * @param key the name of the query parameter (e.g., "api-version", "name", "After").
-     * @param value the value of the query parameter
-     * @return the updated URL with the appended query parameter.
+     * @param host The base URL to which the query parameter will be appended.
+     * @param queryParams A map containing the query parameters and their values.
+     * @param delimiter The delimiter to use if the value is a list.
+     * @return The URL with the appended query parameter.
      */
-    public static String appendQueryParam(String url, String key, String value) {
-        if (value == null) {
-            return url;
+    public static String appendQueryParam(String host, Map<String, Object> queryParams, char delimiter) {
+        if (queryParams == null || queryParams.isEmpty()) {
+            return host;  // No parameters to append
         }
 
-        // Append query parameter to URL
-        if (url.contains("?")) {
-            return url + "&" + key + "=" + value;
-        } else {
-            return url + "?" + key + "=" + value;
+        StringBuilder urlBuilder = new StringBuilder(host);
+        boolean hasExistingQuery = host.contains("?");
+
+        // Process each key-value pair in the queryParams map
+        for (Map.Entry<String, Object> entry : queryParams.entrySet()) {
+            String key = encodeValue(entry.getKey());
+            Object value = entry.getValue();
+            String valueString;
+
+            if (value instanceof List<?>) {
+                List<?> valueList = (List<?>) value;
+                valueString = valueList.stream()
+                    .map(Object::toString)
+                    .map(CoreUtils::encodeValue)  // Encoding each value
+                    .collect(Collectors.joining(String.valueOf(delimiter)));  // Join with delimiter
```

Comment:
The delimiter should be encoded too. It might be best to first create the `valueString` and then call `encode`

Reply:
Doing encoding in a seperate PR.